Fix/RMET-376/mab6.3 android sdk30 build crash

andregrillo opened the pull request:
## Description
<!--- Describe your changes in detail -->
Merge branch **fix/Mab6.3-Android-sdk30-build-crash** into OutSystems branch and keep the custom changes made by OutSystems
## Context
<!--- Place the link to the issue here preceded by either 'Closes' OR 'Fixes' -->
Closes the issue: https://outsystemsrd.atlassian.net/browse/RMET-376

<!--- Why is this change required? What problem does it solve? -->
This change fixes the MABS 6.3 build crash on the Android platform and keeps the MABS 7 build working as before.
 
## Type of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply -->
- [ ] Feature (change which adds functionality)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to new feature)
- [x] Fix (change which fixes an issue)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to fix)
- [ ] Performance (change which improves performance)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to performance improvement)
- [ ] Refactor (non-breaking change that is neither feature, fix nor performance)
- [ ] Style (non-breaking change that only affects formatting and/or white-space)

## Components affected
- [x] Android platform
- [ ] iOS platform
- [ ] JavaScript
- [ ] OutSystems

## Tests
<!--- Describe how you tested your changes in detail -->
Tested with Camera Sample App in sauce labs devices
<!--- Include details of your test environment if relevant -->

## Screenshots (if appropriate)
<!--- E.g. before change & after change -->

## Checklist
<!--- Go over all the following items and put an `x` in all the boxes that apply -->
- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [ ] Tests have been created
- [x] Code follows the code style of this project
- [ ] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

andregrillo committed:
Mabs version checker hook

Detects Mabs version to implement Android SDK 30 for Mabs 7 only.

andregrillo committed:
Hook Call

andregrillo committed:
Hook call changed

andregrillo committed:
Hook call change

andregrillo committed:
Hook fix

andregrillo committed:
utils.js hook updated

cmfsotelo commented on hooks/Android/MABSVersionChecker.js:
```diff
@@ -0,0 +1,37 @@
+                      throw new Error('Camera Plugin: Unable to read config.xml: ' + err);
+                    }
+                    
+                    var result = data.replace(/<platform name="android">/g, '<platform name="android"><config-file target="AndroidManifest.xml" parent="/*"><queries><intent><action android:name="android.media.action.IMAGE_CAPTURE" /></intent><intent><action android:name="android.intent.action.GET_CONTENT" /></intent></queries></config-file>');
```
This might be incompatible with other plugins that add the very same to the config.xml

Can you make some minor validation to check if this is not already in place? 
I believe having double specs breaks the build

cmfsotelo commented on hooks/Android/MABSVersionChecker.js:
```diff
@@ -0,0 +1,37 @@
+var fs = require('fs'), path = require('path');
+var utils = require("./utils");
```
I wonder if this should all be under "hooks/" and not "hooks/Android"

The **_utils._**js file even more, since they seem platform agnostic utils

Also, the hook file "MABSVersionChecker" does not seem to have an appropriate name, as it does't only checks the version, but it actually performs changes.

Maybe something like "PatchAndroidManifest.js"

andregrillo replied:
Good points that I will make the changes.

cmfsotelo reviewed (commented):
Hello,

Minor comments ;)

andregrillo committed:
Hooks Improvements

Created a test to check if the AndroidManifest.xml was already being patched in config.xml and some other minor changes

cmfsotelo reviewed (approved):
LGTM ;)

vmfo22 reviewed (approved):
LGTM
